crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods/mod.rs

```diff
@@ -41,25 +44,139 @@ impl PeepholeReplaceKnownMethods {
 
         let Expression::StaticMemberExpression(member) = &call_expr.callee else { return };
         if let Expression::StringLiteral(string_lit) = &member.object {
-            if call_expr.arguments.len() == 0 {
-                let transformed_value = match member.property.name.as_str() {
-                    "toLowerCase" => Some(
-                        ctx.ast.string_literal(call_expr.span, string_lit.value.cow_to_lowercase()),
-                    ),
-                    "toUpperCase" => Some(
-                        ctx.ast.string_literal(call_expr.span, string_lit.value.cow_to_uppercase()),
-                    ),
-                    "trim" => Some(ctx.ast.string_literal(call_expr.span, string_lit.value.trim())),
-                    _ => None,
-                };
-
-                if let Some(transformed_value) = transformed_value {
-                    self.changed = true;
-                    *node = ctx.ast.expression_from_string_literal(transformed_value);
+            let replacement = match member.property.name.as_str() {
+                "toLowerCase" | "toUpperCase" | "trim" => {
+                    let transformed_value =
+                        match member.property.name.as_str() {
+                            "toLowerCase" => Some(ctx.ast.string_literal(
+                                call_expr.span,
+                                string_lit.value.cow_to_lowercase(),
+                            )),
+                            "toUpperCase" => Some(ctx.ast.string_literal(
+                                call_expr.span,
+                                string_lit.value.cow_to_uppercase(),
+                            )),
+                            "trim" => Some(
+                                ctx.ast.string_literal(call_expr.span, string_lit.value.trim()),
+                            ),
+                            _ => None,
+                        };
+
+                    transformed_value.map(|transformed_value| {
+                        ctx.ast.expression_from_string_literal(transformed_value)
+                    })
                 }
+                "indexOf" | "lastIndexOf" => self.try_fold_string_index_of(
+                    call_expr.span,
+                    call_expr,
+                    member,
+                    string_lit,
+                    ctx,
+                ),
+                "substr" => self.try_fold_string_substr(node, ctx),
+                "substring" | "slice" => self.try_fold_string_substring_or_slice(node, ctx),
+                "charAt" => self.try_fold_string_char_at(node, ctx),
+                "charCodeAt" => self.try_fold_string_char_code_at(node, ctx),
+                "replace" => self.try_fold_string_replace(node, ctx),
+                "replaceAll" => self.try_fold_string_replace_all(node, ctx),
```

Comment:
I recommend you add these functions when implementing it, otherwise it can be annoying to navigate to unimplemented methods.